sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -2190,6 +2190,20 @@ def expand(self, pcoll):
               'A schema must be provided when writing to BigQuery using '
               'Avro based file loads')
 
+      if self.schema and type(self.schema) is dict:
+
+        def find_in_nested_dict(schema):
+          for field in schema['fields']:
+            if field['type'] == 'JSON':
+              raise ValueError(
+                  'Found JSON type in table schema. JSON data '
+                  'insertion is currently not supported with '
+                  'FILE_LOADS write method.')
+            elif field['type'] == 'STRUCT':
+              find_in_nested_dict(field)
```

Comment:
that is correct. I can add the [bq json type doc#batchloads](https://cloud.google.com/bigquery/docs/reference/standard-sql/json-data#use_a_batch_load_job) to the message. Should i also mention `"because Beam doesn't support CSV format"`?

Reply:
hmmm... maybe not : P - wdyt?

Reply:
i don't like the sound of it either, but thought of suggesting because the link doesn't add that nuance (in case the user wants to know why it's not supported yet).

we could just mention that it is supported for the other write methods and put the link for more details?

Reply:
right - yeah, let's do that!